### PR TITLE
[Website] Partners schema

### PR DESCRIPTION
**Pull request by FaithDaka**
#### Summary of Changes (What does this PR do?)
- Features partners schema to allow for dynamic addition of content.
- Fix for window history in two frontend components.

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* cd website
* git pull origin staging
* git checkout origin/ft-partners-schema

In terminal 1
* npm run build
* npm run serve

In terminal 2 (create virtual environment)
* python manage.py migrate
* python manage.py collectatic
* python manage.py runserver

Route to http://127.0.0.1:8000/admin/ to view table

#### What are the relevant tickets?
- [WEB-328]

#### Screenshots 
Screenshot (53)

[WEB-328]: https://airqoteam.atlassian.net/browse/WEB-328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/backend/partners/admin.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    logo_preview.allow_tags = True
+
+    def image_preview(self, obj):
+        width, height = 120, 80
+        from django.utils.html import escape, format_html
+
+        if obj.partner_image:
+            return format_html(f'<img src="{escape(obj.partner_image.url)}" width="{width}" height="{height}" />')
+
+    image_preview.allow_tags = True
```
Comment (FaithDaka) [inline website/backend/partners/admin.py:38]: Image view specifications

File: website/backend/partners/models.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    partner_image = CloudinaryField("PartnerImage", overwrite=True, resource_type="image", null=True, blank=True)
+    partner_logo = CloudinaryField("PartnerLogo", overwrite=True, resource_type="image")
+    partner_name = models.CharField(max_length=200, null=False, blank=False)
+    order = models.IntegerField(default=1)
+    partner_link = models.URLField(null=True, blank=True)
+    unique_title = models.CharField(max_length=100, null=True, blank=True)
+    type = models.CharField(
+        max_length=40, default=RelationTypes.Partnership, choices=RelationTypes.choices, null=True, blank=True
+    )
```
Comment (FaithDaka) [inline website/backend/partners/models.py:24]: Partner object model

File: website/backend/partners/models.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    class Meta:
+        ordering = ['order', 'id']
```
Comment (FaithDaka) [inline website/backend/partners/models.py:27]: Order objects by order number and ascending order of IDs

File: website/backend/partners/models.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            return self.generate_unique_title(postfix_index=postfix_index)
+
+    def __str__(self):
+        return f"Partner - {self.partner_name}"
+
+
+@receiver(pre_save, dispatch_uid="append_short_name", sender=Partner)
+def append_short_name(sender, instance, *args, **kwargs):
+    if not instance.unique_title:
+        instance.unique_title = instance.generate_unique_title()
```
Comment (FaithDaka) [inline website/backend/partners/models.py:51]: Unique title for frontend URL creation

File: website/backend/partners/models.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+class PartnerDescription(BaseModel):
+    description = models.TextField(null=True, blank=True)
+    order = models.IntegerField(default=1)
+    partner = models.ForeignKey(
+        Partner,
+        null=True,
+        blank=True,
+        related_name="descriptions",
+        on_delete=models.deletion.SET_NULL,
+    )
```
Comment (FaithDaka) [inline website/backend/partners/models.py:64]: Partner description model

File: website/backend/urls.py [hunk 6]
```diff
@@ -40,6 +41,7 @@
 api_router.register(r'highlights', highlight_views.HighlightViewSet)
 api_router.register(r'departments', career_views.DepartmentViewSet)
 api_router.register(r'tags', tag_views.TagViewSet)
+api_router.register(r'partner', partner_views.PartnerViewSet)
```
Comment (FaithDaka) [inline website/backend/urls.py:44]: Added partner URLs to API

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @FaithDaka 
LGTM

File: website/backend/partners/tests.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
```
Comment (Baalmart) [inline website/backend/partners/tests.py:2]: Great that we getting setup for writing unit tests. Brilliant!

**Review by Baalmart — APPROVED** [review]
LGTM, thanks @FaithDaka

**Pull request merged**